integration/spark-common/src/main/java/org/apache/carbondata/spark/merger/CarbonCompactionUtil.java

```diff
@@ -351,4 +351,33 @@ private static int getDimensionDefaultCardinality(CarbonDimension dimension) {
     }
     return cardinality;
   }
+
+  /**
+   * This method will check for any restructured block in the blocks selected for compaction
+   *
+   * @param segmentMapping
+   * @param dataFileMetadataSegMapping
+   * @param tableLastUpdatedTime
+   * @return
+   */
+  public static boolean checkIfAnyRestructuredBlockExists(Map<String, TaskBlockInfo> segmentMapping,
+      Map<String, List<DataFileFooter>> dataFileMetadataSegMapping, long tableLastUpdatedTime) {
+    boolean restructuredBlockExists = false;
+    for (Map.Entry<String, TaskBlockInfo> taskMap : segmentMapping.entrySet()) {
+      String segmentId = taskMap.getKey();
+      List<DataFileFooter> listMetadata = dataFileMetadataSegMapping.get(segmentId);
+      for (DataFileFooter dataFileFooter : listMetadata) {
+        // if schema modified timestamp is greater than footer stored schema timestamp,
```

Comment:
yes...because the entry will be added in schema evolution entry and in case of any failure we need to revert back the schema